### PR TITLE
Add APIs: get film by ID, delete film

### PR DESCRIPTION
**Pull request by qhuongng**

### CODE REVIEW

File: week2/internal/wire.go [hunk 1]
```diff
@@ -24,15 +24,15 @@ var serverSet = wire.NewSet(
 
 // handler === controller | with service and repository layers to form 3 layers architecture
 var handlerSet = wire.NewSet(
-	v1.NewActorHandler,
+	v1.NewActorHandler, v1.NewFilmHandler,
```
Comment (VuKhoa23) [inline week2/internal/wire.go]: should declare in new line
```
v1.NewActorHandler, 
v1.NewFilmHandler,
```

File: week2/internal/wire.go [hunk 2]
```diff
@@ -24,15 +24,15 @@ var serverSet = wire.NewSet(
...
 )
 
 var serviceSet = wire.NewSet(
-	serviceimplement.NewActorService,
+	serviceimplement.NewActorService, serviceimplement.NewFilmService,
```
Comment (VuKhoa23) [inline week2/internal/wire.go]: should declare in new line
```
serviceimplement.NewActorService, 
serviceimplement.NewFilmService,
```

File: week2/internal/wire.go [hunk 3]
```diff
@@ -24,15 +24,15 @@ var serverSet = wire.NewSet(
...
 )
 
 var repositorySet = wire.NewSet(
-	repositoryimplement.NewActorRepository,
+	repositoryimplement.NewActorRepository, repositoryimplement.NewFilmRepository,
```
Comment (VuKhoa23) [inline week2/internal/wire.go]: should declare in new line
```
repositoryimplement.NewActorRepository, 
repositoryimplement.NewFilmRepository,
```

File: week2/internal/domain/model/film.go [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package model
+
+type FilmRequest struct {
```
Comment (VuKhoa23) [inline week2/internal/domain/model/film.go]: not sure what this struct for

File: week2/internal/controller/http/v1/film_handler.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	parsedId, err := strconv.ParseInt(id, 10, 64)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, model.HttpResponse[any]{Message: "Invalid film ID", Data: nil})
+		return
+	}
+
+	err = handler.filmService.DeleteFilm(c.Request.Context(), parsedId)
+	if err != nil {
+		if err.Error() == "record not found" {
+			c.JSON(http.StatusNotFound, model.HttpResponse[any]{Message: "Film not found", Data: nil})
```
Comment (VuKhoa23) [inline week2/internal/controller/http/v1/film_handler.go]: I think status code should be 400

**Review by VuKhoa23 — CHANGES_REQUESTED** [review]

File: week2/.env.template
Comment (VuKhoa23) [inline week2/.env.template:1]: Why delete this file :v

**Review by VuKhoa23 — CHANGES_REQUESTED** [review]

File: week2/.env.template
Comment (VuKhoa23) [inline week2/.env.template:1]: why delete this file :v

File: week2/internal/repository/implement/film_repository.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	return &film, nil
+}
+
+func (repo *FilmRepository) DeleteFilm(ctx context.Context, id int64) error {
+	return repo.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
+		// Delete from film_actor
+		if err := tx.Exec(`
+            DELETE fa FROM film_actor fa
+            JOIN film f ON f.film_id = fa.film_id
+            WHERE f.film_id = ?`, id).Error; err != nil {
```
Comment (VuKhoa23) [inline week2/internal/repository/implement/film_repository.go]: Not sure why we need to JOIN here. Can we just delete like this
```
DELETE fa FROM film_actor fa
WHERE fa.film_id = ?
```
And the same for below queries

File: week2/internal/domain/model/film.go (hunk 4 above)
Comment (qhuongng) [reply week2/internal/domain/model/film.go]: Me either, I just figured we might need it later. Removed :v

File: week2/.env.template
Comment (qhuongng) [reply week2/.env.template:1]: I renamed .env.template to .env. Added back

File: week2/internal/controller/http/v1/film_handler.go (hunk 5 above)
Comment (VuKhoa23) [reply week2/internal/controller/http/v1/film_handler.go]: What is your opinion on this @qhuongng

File: week2/internal/repository/implement/film_repository.go (hunk 6 above)
Comment (qhuongng) [reply week2/internal/repository/implement/film_repository.go]: Generating a DELETE statement from MySQL resulted in queries with those JOINs baked in so I was just being extra careful by copying it 1:1

File: week2/internal/controller/http/v1/film_handler.go (hunk 5 above)
Comment (qhuongng) [reply week2/internal/controller/http/v1/film_handler.go]: StackOverflow suggested 204 No Content but that was also highly debatable, so I think using 400 is appropriate for now

File: week2/internal/repository/implement/film_repository.go (hunk 6 above)
Comment (VuKhoa23) [reply week2/internal/repository/implement/film_repository.go]: I think this may affect to the performance and readability of these delete queries

Comment (qhuongng) [reply week2/internal/repository/implement/film_repository.go]: I'll try removing the JOINs and see what happens

Comment (qhuongng) [reply week2/internal/repository/implement/film_repository.go]: Can confirm it works fine. Resolved

**Review by VuKhoa23 — APPROVED** [review]
LGTM !

**Pull request merged**